import { Component, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';

@Component({
  selector: 'app-tool',
  templateUrl: './tool.component.html',
  styleUrls: ['./tool.component.css']
})
export class ToolComponent implements OnInit {
  heading = 'tools';

  toolname: string
  tooldescription: string
  tooldisplayname: string
  toolpic: string
  toollink: string

  public tools:Map<string, Array<string>> = new Map([["Proper", ["Proper", "PROPER, or PROspective Power Evaluation for RNAseq, is a freely available R package for power assessment and sample size determination for RNA-seq. PROPER includes two components: (i) a semi-parametric simulation that generates data based on actual RNA-seq experiments with flexible choices on baseline expressions, biological variations and patterns of DE; and (ii) a power assessment component that provides a comprehensive view of power.", "https://www.bioconductor.org/packages/release/bioc/html/PROPER.html", "https://bioconductor.org/images/logo_bioconductor.gif"]], 
  	["AnalyzeReplication", ["AnalyzeReplication", "AnalyzeReplication is a low-cost metric for guiding the design of large-scale RNA-seq Differential Expression (DE) studies. AnalyzeReplication is able to improve a single target: power to detect DE genes between samples.", "http://home.uchicago.edu/~jiezhou/replication/", "https://d3qi0qp55mx5f5.cloudfront.net/shared-resources/i/template/uc_wordmark_hires.gif"]],
  	["RNAtor", ["RNAtor", "RNAtor is a biologist-friendly and easy-to-use Android platform to design RNA-seq experiments based on certain user inputs. Where sample is limiting, RNAtor provides guidelines to produce required number of reads to detect differentially expressed transcripts. The recommendations provided by RNAtor are based on an exhaustive combination of simulation studies and validation with real RNA-seq datasets. RNAtor is available on the Google Play Store.", "https://github.com/binaypanda/RNAtor", "https://f1000researchdata.s3.amazonaws.com/manuscripts/12955/51533580-554c-43f7-9e5c-f97c08f4f457_figure1.gif"]],
  	["BwB", ["BioDepot-workflow-Builder (BwB)", "BwB, or BioDepot-workflow-Builder, provides a user-friendly, modular and open-source graphical user interface (GUI) to create bioinformatics workflows. As an additional feature, the GUI is also wrapped inside a Docker container and can be accessed through most major web browsers. The use of software containers ensures that workflows will give identical results across different operating systems and hardware architectures. The use of Docker also allows for individual components to be deployed on the cloud. The modularity and ease of customization and installation of bioinformatics tools using BwB allows for researchers to efficiently test new workflows and compare competing algorithms. This means that the only requirement for creating and running BioDepot workflows is to install Docker. BwB can be used to build different workflows by combining interchangeable and encapsulated widgets, allowing researchers to easily test alternative algorithms and observe how the outputs differ.", "https://github.com/BioDepot/BioDepot-workflow-builder", "https://github.com/BioDepot/BioDepot-workflow-builder/blob/master/Media/logo.png?raw=true"]],
  	["Alta-cyclic", ["Alta-cyclic", "Alta-cyclic is an Illumina Genome-Analyzer (Solexa) base caller. Alta-Cyclic uses machine learning to compensate for noise factors. Alta-Cyclic substantially improved the number of accurate reads for sequencing runs up to 78 bases and reduced systematic biases, facilitating confident identification of sequence variants. Alta-Cyclic has several notable differences from the standard Illumina base caller that should improve read accuracy. First, all the calling parameters are optimized empirically and tested to enhance the accuracy of the base calling for each run, whereas in the Illumina base caller parameters are statically derived, though in a sophisticated manner, and are not evaluated. Second, Alta-Cyclic calculates phasing parameters based on a parametric model using data from the latest cycles, whereas the Illumina base caller relies on a numerical method and data from early cycles. Finally, Alta-Cyclic dynamically tracks changes in fluorophore cross-talk, which severely disrupts signals in later cycles, whereas the Illumina base caller statically determines cross-talk based on values from early cycles.", "http://hannonlab.cshl.edu/Alta-Cyclic/main.html", "http://slideplayer.com/5799910/19/images/12/Base+calling%3A+Alta-Cyclic.jpg"]],
  	["FlowgramFixer", ["FlowgramFixer", "FlowgramFixer is a freely available,  improved basecaller for the IonTorrent sequencing platform, which reduces error rates, and generates more uniquely aligned reads and more high quality reads than the default base calling algorithm implemented by IonTorrent in TorrentSuite. FlowgramFixer is also applicable to others flowgram-based sequencing platforms. FlowgramFixerâ€™s showcases its superior performance on Ion Torrent Escherichia coli data, with a 4.8% improvement in the number of high-quality mapped reads and a 7.1% improvement in the number of uniquely mappable reads.", "https://academic.oup.com/bioinformatics/article/29/13/i344/188472#xref-corresp-1-1", "https://www.aha.io/assets/github.7433692cabbfa132f34adb034e7909fa.png"]],
  	["DeepNano", ["DeepNano", "Provides a more accurate and computationally efficient alternative to the Hidden Markov-Model (HMM)-based methods used in the Metrichor base caller by the device manufacturer. DeepNano is a new, open source tool for base calling MinION sequencing data. The MinION device is currently the smallest high-throughput DNA sequencer. DeepNano is  based on deep recurrent neural networks and show that the accuracy of base calling is much dependent on the underlying software and can be improved by considering modern machine learning methods. By employing carefully crafted recurrent neural networks, our tool significantly improves base calling accuracy on data from R7.3 version of the platform compared to the default base caller supplied by the manufacturer.", "http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0178751", "http://brand.cornell.edu/assets/images/examples/trademarks/brand_registered.svg"]],
  	["PyroBayes", ["PyroBayes", "A base caller for pyrosequences from the 454 Life Sciences sequencing machines. PyroBayes was designed to assign more accurate base quality estimates to the 454 pyrosequences. Pyrobayes permits accurate single-nucleotide polymorphism (SNP) calling in resequencing applications, even in shallow read coverage, primarily because it produces more confident base calls than the native base calling program. The increased accuracy of the base qualities will permit more sensitive biological studies using the 454 machines.", "https://github.com/alces-software/packager-base/tree/master/ext/pyrobayes", "https://urgi.versailles.inra.fr/var/storage/images/home-urgi/tools/pyrobayes/17864-1-eng-GB/PyroBayes_medium.jpg"]],
  	["FastQC", ["FastQC", "FastQC aims to provide a simple way to do some quality control checks on raw sequence data coming from high throughput sequencing pipelines. It provides a modular set of analyses which you can use to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis. FastQC can be run in one of two modes. It can either run as a stand alone interactive application for the immediate analysis of small numbers of FastQ files, or it can be run in a non-interactive mode where it would be suitable for integrating into a larger analysis pipeline for the systematic processing of large numbers of files.", "https://www.bioinformatics.babraham.ac.uk/projects/fastqc/", "https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc.png"]],
  	["TrimGalore", ["TrimGalore!", "TrimGalore! is a wrapper script to automate quality and adapter trimming as well as quality control, with some added functionality to remove biased methylation positions for RRBS sequence files (for directional, non-directional (or paired-end) sequencing). When dealing with paired end files, this program more efficient than others in handling such files. TrimGalore! makes use of the publicly available adapter trimming tool Cutadapt and FastQC for optional quality control once the trimming process has completed.", "https://github.com/FelixKrueger/TrimGalore", "https://www.bioinformatics.babraham.ac.uk/images/babraham_bioinformatics.gif"]],
  	["SMART", ["S-MART", "S-MART performs two categories of tasks: data manipulation and visualization. Manipulations include  selection/exclusion of the reads which overlap with some reference sets,  read clustering, and differential expression analysis. Visualizations include read-size distribution, nucleotidic compositions,  chromosome localizations, and distances with respect to a reference set. S-Mart also provides many ways to visualize the data: size of the reads, density of the genome, distance with respect to a reference set, and the correlation of two data sets. S-MART is intuitive and easy to use, even for people with no computer-science background.", "https://urgi.versailles.inra.fr/Tools/S-Mart", "https://urgi.versailles.inra.fr/extension/inra/design/urgi/images/LogoSite.jpg"]],
  	["clcgenomicsworkbench", ["CLC Genomics Workbench", "CLC Genomics Workbench offers a complete and customizable solution for genomics, transcriptomics, epigenomics, and metagenomics. The software enables to generate custom workflows, which can combine quality control steps, adapter trimming, read mapping, variant detection, and multiple filtering and annotation steps into a pipeline. The software features bioinformatic algorithms that are widely used by industry and academia to overcome challenges associated with large-scale data analysis. CLC Genomics Workbench is a product of Qiagen Bioinformatics", "https://www.qiagenbioinformatics.com/products/clc-genomics-workbench/", "https://www.qiagenbioinformatics.com/wp-content/uploads/2017/11/Gx_Nov_2017_top.jpg"]],
  	["BitSeq", ["BitSeq", "There are two stages in BitSeq analysis. The first stage is expression estimation and the second is differential expression estimation. The first stage involves using Bayesian inference to infer expression of each transcript from each RNA-seq experiment. The goal of the second stage is to choose transcripts with the greatest probability of differential expression.", "http://bioconductor.org/packages/release/bioc/html/BitSeq.html", "http://bioconductor.org/images/logo_bioconductor.gif"]],
  	["DesSeq", ["DesSeq", "DESeq is a very powerful and fast R package which uses the negative binomial model in order to test for differentially expressed genes.", "http://bioconductor.org/packages/release/bioc/html/DESeq2.html", "https://bioconductor.org/images/logo_bioconductor.gif"]],
  	["Cufflinks", ["Cufflinks", "Cufflinks is an open source software program that uses algorithms to assemble transcripts, estimate prevalence of these transcripts and also test for differentially expressed genes and regulation in RNA-seq samples.", "http://cole-trapnell-lab.github.io/cufflinks/", "https://www.aha.io/assets/github.7433692cabbfa132f34adb034e7909fa.png"]],
  	["RNAeXpress", ["RNAeXpress", "RNAeXpress is a useful tool used for extraction and annotation of important biological transcripts of RNA-seq data.", "https://rnaexpress.erc.monash.edu.au/", "https://rnaexpress.erc.monash.edu.au/images/frontpage.png"]],
  	["SeqMonk", ["SeqMonk", "SeqMonk is a program used for visualization and analysis of large mapped genomic regions. SeqMonk is designed to work with data from high throughput sequencing technologies but also works with any set of mapped reads. With SeqMonk a genome browser is provided which allows the user quick navigation and mapping of sequence reads to their appropriate genomic context.", "https://www.bioinformatics.babraham.ac.uk/projects/seqmonk/Help/1%20Introduction/1.1%20What%20is%20SeqMonk.html", "https://www.bioinformatics.babraham.ac.uk/images/babraham_bioinformatics.gif"]],
  	["RSEM", ["RSEM", "RSEM is a software package used for quantifying gene and isoform abundances from single-end/ paired-end RNA-Seq data. RSEM does not require using a reference genome and is very useful for quantification with de novo transcriptome assemblies.", "https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323", "https://www.biomedcentral.com/static/images/bmc/logo-bmc-white-small-flagship-d1d064645b.svg"]],
  	["Bowtie", ["Bowtie", "Bowtie is a very fast and efficient short read aligner that is very commonly used. Bowtie utilizes the Burrows-Wheeler index in order to keep memory efficiency.", "http://bowtie-bio.sourceforge.net/index.shtml", "http://bowtie-bio.sourceforge.net/images/bowtie_logo.png"]],
  	["BWA", ["Burrows-Wheeler Aligner (BWA)", "BWA is a software package similar to BOWTIE. BWA maps low-divergent sequences against a large reference genome. It is made up of three algorithms. The first is used for sequence reads of lengths 70bp - 100bp and the other two are used for longer sequences of lengths 70bp - 1Mbp.", "http://bio-bwa.sourceforge.net/", "https://image.slidesharecdn.com/bdt308-131202184341-phpapp02/95/a-step-to-the-clouded-solution-of-scalable-clinical-genome-sequencing-bdt308-aws-reinvent-2013-40-638.jpg?cb=1434517967"]],
  	["BLAST", ["BLAST", "BLAST is a very popular tool used for finding similar regions between biological sequences by comparing amino acid sequences of proteins or nucleotide sequences of DNA. BLAST uses a heuristic algorithm which makes it a lot faster than other tools.", "https://blast.ncbi.nlm.nih.gov/Blast.cgi", "https://blast.ncbi.nlm.nih.gov/images/nucleutide-blast-cover.png"]],
  	["AStalavista", ["AStalavista", "AStalavista (alternative splicing transcriptional landscape visualization tool) dynamically identifies, extracts, and displays alternative splicing events from whole genome annotations and user provided gene sets. It supports alternative transcriptional landscape visualization. In addition, the tool remains applicable even if sequencing/annotation has not been completed.", "https://bioinformatics.ca/links_directory/tool/9250/astalavista", "https://bioinformatics.ca/links_directory/files/system-generated/bioinformatics_next_logo.png"]],
  	["AltAnalyze", ["AltAnalyze", "AltAnalyze can assess alternative exon expression along protein isoforms, domain composition, and microRNA targeting. Is can also provide various methods for the analysis of other data, like RNA summarization, annotation, and clustering. In addition, it can serve as an application for microarray, RNA-seq, and metabolomics analysis.", "http://www.altanalyze.org", "http://www.altanalyze.org/graphics/logoX1.gif"]],
  	["spliceR", ["spliceR", "spliceR is an R package for classification of alternative splicing and prediction of coding potential. The tool is able to detect coding potential of transcripts, determine untranslated region and open reading frame lengths, and predict whether transcripts are nonsense mediated decay-sensitive based on compatible annotated start codon positions and their downstream open reading frame.", "https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-81", "https://bmcbioinformatics.biomedcentral.com/static/images/bmc/logo-bmc-white-small-series-cb0674b77f.svg"]],
  	["maSigPro", ["maSigPro", "maSigPro is an R package that uses a regression based approach to find genes for which there are significant gene expression profile differences between experimental groups in time course microarray and RNA-seq experiments. It also allows for the analysis of multiple time course transcriptomics data.", "https://bioconductor.org/packages/release/bioc/html/maSigPro.html", "https://bioconductor.org/images/logo_bioconductor.gif"]],
  	["diffsplicing", ["diffsplicing", "diffsplicing is an R package that tests for differential exon usage for each gene. Ir models expression levels in three different settings: overall gene expression level, absolute transcript expression level and relative transcript expression level expressed as a proportion of all transcripts for the same gene. In addition, it ranks the genes and transcripts according to the temporal change they show in their expression levels.", "http://web.mit.edu/~r/current/arch/i386_linux26/lib/R/library/limma/html/diffSplice.html", "http://www.netlab.uky.edu/p/bioinfo/sites/www.netlab.uky.edu.p.bioinfo/files/sky_logo.png"]],
  	["timeSeq", ["timeSeq", "imeSeq is an R package for nonparallel differential expression genes and parallel differential expression genes detection. The tool uses a binomial mixed-effect model. In addition, is includes a parallel computing option to speed up the computing process.", "https://www.rdocumentation.org/packages/timeSeq/versions/1.0.3/topics/timeSeq", "https://rud.is/b/wp-content/uploads/2016/02/RStudio.png"]],
  	["LIONS", ["LIONS", "LIONS is an analysis pipeline which is used to annotates a paired-end RNAseq library to detect transposable element transcripts. It quantifies the contribution of transposable element promoters to the transcriptome from the RNA seq data.", "https://www.biorxiv.org/content/early/2017/06/13/149864", "https://www.biorxiv.org/sites/default/files/bioarchives_article%232514A7F.jpg"]],
  	["TEtools", ["TEtools", "TEtools is used for the analysis of transposable element expression. It works by taking into account the transposable element sequence diversity of the genome, and can be applied to the unannotated or unassembled genome. It also performs mapping of RNA-seq data from classical mRNAs or small RNAs onto a list of transposable element sequences.", "https://academic.oup.com/nar/article/45/4/e17/2290912", "http://www.rna-seqblog.com/wp-content/uploads/2016/12/tetools.jpg"]],
  	["RepEnrich", ["RepEnrich", "RepEnrich is a computational method used to study genome-wide transcriptional regulation of repetitive elements, like transposable elements. It also supports custom repetitive element or repeat feature annotation in bed format.", "https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4757950", "https://kanbanize.com/blog/wp-content/uploads/2014/11/GitHub-300x175.jpg"]]]);

  constructor(private route: ActivatedRoute) {
    this.toolname = route.snapshot.params['id'];
    	this.tooldescription = this.tools.get(this.toolname)[1];
    	this.tooldisplayname = this.tools.get(this.toolname)[0];
    	this.toolpic = this.tools.get(this.toolname)[3];
    	this.toollink = this.tools.get(this.toolname)[2];
}
  

  ngOnInit() {
    
  }

}
